server/server.go

```diff
@@ -164,7 +164,7 @@ func (s *Server) HandleReview(ctx context.Context, e *lookout.ReviewEvent, safeP
 		if err != nil {
 			return nil, err
 		}
-		return resp.Comments, nil
+		return s.dedupComments(resp.Comments), nil
```

Comment:
maybe it's better to deduplicate comments from all analyzers?

Reply:
I don't know. Anyway I miss some context, is the final user aware of which analyzer generated the comment? Or do they appear under the same lookout bot?

Because in general I'd say that there's no difference for the same comment being generated by differente analyzer, hence dedupping on the whole set of comments sounds good. But could be the final user interested in for example being notified about all comments of all analyzer, but addressing only those of some analyzers? In this scenario the user would want to have all comments and the deduplication should be applied at single analyzer level. Obviously this doesn't apply if everything comes out as posted by a single bot.

Reply:
Currently, everything is posted by the same single bot. And there is no way to distinguish which analyzer produced the comment from a user point of view except link at the footer:
<img width="679" alt="screenshot 2019-01-10 at 18 07 58" src="https://user-images.githubusercontent.com/406916/50984684-b908f280-1502-11e9-8612-b7bf4a94d015.png">

I'm also not sure if we should deduplicate all the comments but it is what looks more logically correct for me.

Reply:
Given that there's no way to distinguish which analyzer produced the comment I agree that deduplicating all the comments is more consistent.